cpp/oneapi/dal/backend/primitives/rng/dpc_engine.hpp

```diff
@@ -0,0 +1,233 @@
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/primitives/rng/utils.hpp"
+#include "oneapi/dal/backend/primitives/rng/rng_types.hpp"
+
+#include <daal/include/algorithms/engines/mt2203/mt2203.h>
+#include <daal/include/algorithms/engines/mcg59/mcg59.h>
+#include <daal/include/algorithms/engines/mrg32k3a/mrg32k3a.h>
+#include <daal/include/algorithms/engines/philox4x32x10/philox4x32x10.h>
+#include <daal/include/algorithms/engines/mt19937/mt19937.h>
+
+#include <oneapi/mkl.hpp>
+
+namespace mkl = oneapi::mkl;
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+template <engine_method EngineType>
+struct dpc_engine_type;
+
+template <>
+struct dpc_engine_type<engine_method::mt2203> {
+    using type = oneapi::mkl::rng::mt2203;
+};
+
+template <>
+struct dpc_engine_type<engine_method::mcg59> {
+    using type = oneapi::mkl::rng::mcg59;
+};
+
+template <>
+struct dpc_engine_type<engine_method::mt19937> {
+    using type = oneapi::mkl::rng::mt19937;
+};
+
+template <>
+struct dpc_engine_type<engine_method::mrg32k3a> {
+    using type = oneapi::mkl::rng::mrg32k3a;
+};
+
+template <>
+struct dpc_engine_type<engine_method::philox4x32x10> {
+    using type = oneapi::mkl::rng::philox4x32x10;
+};
+
+/// A class that provides a unified interface for random number generation on both CPU and GPU devices.
+///
+/// This class serves as a wrapper for random number generators (RNGs) that supports different engine types,
+/// enabling efficient random number generation on heterogeneous platforms using SYCL. It integrates a host
+/// (CPU) engine and a device (GPU) engine, allowing operations to be executed seamlessly on the appropriate
+/// device.
+///
+/// @tparam EngineType The RNG engine type to be used. Defaults to `engine_method::mt2203`.
+///
+/// @param[in] queue The SYCL queue used to manage device operations.
+/// @param[in] seed  The initial seed for the random number generator. Defaults to `777`.
```

Comment:
Please move those to the constructor's description.

Reply:
It would be also good to provide the descriptions to every public method, not only to the class.